src/pipeline/src/etl/processor/gsub.rs

```diff
@@ -0,0 +1,229 @@
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use regex::Regex;
+
+use crate::etl::field::{Field, Fields};
+use crate::etl::processor::{
+    yaml_bool, yaml_field, yaml_fields, yaml_string, Processor, FIELDS_NAME, FIELD_NAME,
+    IGNORE_MISSING_NAME,
+};
+use crate::etl::value::{Array, Map, Value};
+
+pub(crate) const PROCESSOR_GSUB: &str = "gsub";
+
+const REPLACEMENT_NAME: &str = "replacement";
+const PATTERN_NAME: &str = "pattern";
+
+/// only support string value, and array string value
+#[derive(Debug, Default)]
+pub struct GsubProcessor {
+    fields: Fields,
+    pattern: Option<Regex>,
+    replacement: Option<String>,
+    ignore_missing: bool,
+}
+
+impl GsubProcessor {
+    fn with_fields(&mut self, fields: Fields) {
+        self.fields = fields;
+    }
+
+    fn with_ignore_missing(&mut self, ignore_missing: bool) {
+        self.ignore_missing = ignore_missing;
+    }
+
+    fn try_pattern(&mut self, pattern: &str) -> Result<(), String> {
+        self.pattern = Some(Regex::new(pattern).map_err(|e| e.to_string())?);
+        Ok(())
+    }
+
+    fn with_replacement(&mut self, replacement: impl Into<String>) {
+        self.replacement = Some(replacement.into());
+    }
+
+    fn check(self) -> Result<Self, String> {
+        if self.pattern.is_none() {
+            return Err("pattern is required".to_string());
+        }
+
+        if self.replacement.is_none() {
+            return Err("replacement is required".to_string());
+        }
+
+        Ok(self)
+    }
+
+    fn process_string_field(&self, val: &str, field: &Field) -> Result<Map, String> {
+        let replacement = self.replacement.as_ref().unwrap();
+        let new_val = self
+            .pattern
+            .as_ref()
+            .unwrap()
+            .replace_all(val, replacement)
+            .to_string();
+        let val = Value::String(new_val);
+
+        let key = match field.target_field {
+            Some(ref target_field) => target_field,
+            None => field.get_field(),
+        };
+
+        Ok(Map::one(key, val))
+    }
+
+    fn process_array_field(&self, arr: &Array, field: &Field) -> Result<Map, String> {
+        let key = match field.target_field {
+            Some(ref target_field) => target_field,
+            None => field.get_field(),
+        };
+
+        let re = self.pattern.as_ref().unwrap();
+        let replacement = self.replacement.as_ref().unwrap();
+
+        let mut result = Array::default();
```

Comment:
Not sure current implementation. But looks like the Array can be replaced with Arrow's Array, and use vector execution to process these transforms. It's good for performance.

Reply:
Currently pipeline use its own `Value` enum(including an `Array(Array)` to hold array of `Value`). We will change its value type in next major refactor to boost performance.